pkgs/os-specific/darwin/apple-source-releases/adv_cmds/package.nix

```diff
@@ -76,25 +112,53 @@ mkAppleDerivation {
 
   nativeBuildInputs = [
     bison
+    bmake
     flex
     pkg-config
   ];
 
+  enableParallelBuilding = true;
+
   mesonFlags = [
     # Even though adv_cmds is built with a newer SDK, the default SDK is still the deployment target.
     # Don’t build packages that use newer APIs unnecessarily.
     (lib.mesonOption "sdk_version" (lib.getVersion apple-sdk))
   ];
 
-  postBuild = ''
-    # Build the locales TODO
-  '';
+  buildPhase =
+    ''
+      ninjaBuildPhase
+    ''
```

Comment:
This is missing the `preBuild`/`postBuild` hooks. Is there a way to just put the `bmake` stuff in `postBuild`, or maybe we can make the hooks compose better somehow?

Reply:
`ninjaBuildPhase` and `bmakeBuildPhase` run the hooks. Putting `bmakeBuildPhase` in `postBuild` causes it to go into an infinite loop. I want the flag handling of both hooks without their invoking `preBuild` and `postBuild` hooks, but there’s no way to do that currently.